add new "exclude count" page number pagination option

aleontiev opened the pull request:
...to support a faster page-number pagination over large datasets

- Like standard `PageNumberPagination`, except the results are never counted
- Uses the N+1 trick from `CursorPagination` to determine whether the dataset has more pages
- Can be enabled on any request via query parameter `exclude_count` (name can be configured)

TODO: 
- [x] ensure all current tests pass
- [x] add new test case with `exclude_count` enabled

chdastolfo reviewed (approved):
LGTM post version bump

suavesav commented on dynamic_rest/paginator.py:
```diff
@@ -0,0 +1,96 @@
+
+    def __init__(self, object_list, per_page, orphans=0,
+                 allow_empty_first_page=True, exclude_count=False):
+        self.object_list = object_list
+        self._check_object_list_is_ordered()
+        self.per_page = int(per_page)
+        self.orphans = int(orphans)
+        self.allow_empty_first_page = allow_empty_first_page
```
can we just call the super class' __init__ to do this?

aleontiev replied:
Good point

aleontiev replied:
done

suavesav commented on dynamic_rest/paginator.py:
```diff
@@ -0,0 +1,96 @@
+        self.allow_empty_first_page = allow_empty_first_page
+        self.exclude_count = exclude_count
+
+    def validate_number(self, number):
```
what version of django did you use to get these method's django defn? maybe we should add that info here as a comment?

aleontiev replied:
This was taken from 3.1, but it seems to work across all tested versions in DREST tox (from 2.2 onwards)

aleontiev replied:
done

suavesav commented on dynamic_rest/paginator.py:
```diff
@@ -0,0 +1,96 @@
+            # to determine if more pages are available
+            top = top + 1
+        else:
+            # skip validating against count
```
this comment line should be in the `if` part of this conditional or can be removed altogether

aleontiev replied:
done

suavesav reviewed (commented):
Couple of questions about DynamicPaginator

aleontiev commented:
@chdastolfo @suavesav 
- all tests pass including the new one I added 
- addressed all PR feedback
- bumped version to `2.1.3` in setup.py

so I am going to merge 🚀
